nixos/modules/services/system/guix-daemon.nix

```
@@ -0,0 +1,72 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  cfg = config.guix;
+in {
+  options = with lib; {
+    guix = {
+      enable = mkEnableOption "guix";
+
+      package = mkOption {
+        type = with types; package;
+        default = pkgs.guix;
+        defaultText = literalExpression "pkgs.guix";
+        description = mdDoc ''
+          This option specifies the Guix package to be used in the system.
+        '';
+      };
+
+      substituteUrls = mkOption {
+        type = with types; listOf str;
+        default = [
+          "https://ci.guix.gnu.org"
+          "https://bordeaux.guix.gnu.org"
+        ];
+        description = mdDoc ''
+          List of urls to be used as substituters.
+        '';
+      };
+
+      cores = mkOption {
+        type = with types; int;
```

Using `with types;` for everything is a bit overkill. Would you mind using `types.` when it's a simple type?

```suggestion
        type = types.ints.unsigned;
```